src/System.Windows.Forms/src/System/Windows/Forms/AccessibleObject.cs

```
@@ -350,7 +350,7 @@ internal virtual bool IsPatternSupported(UiaCore.UIA patternId)
             return false;
         }
 
-        internal virtual int[]? RuntimeId => null;
+        internal virtual int[] RuntimeId => throw new NotSupportedException();
```

Why not?
```suggestion
        internal virtual int[] RuntimeId => Array.Empty<int>();
```

---

This would hide the developer error.

I would prefer an exception as it sounds like it is the most noticeable option, visible in the debug output, instead of silent unsupported property.

---

Makes sense

---

If I understand correctly, end user see no difference in all of these cases. Developer scenarios are impacted by throwing exception and Debug statement. Given that throwing exception slows the inspect tool, why not just go Debug statements?

---

Exceptions will show in the debug output, so it is visible to the developer.

---

Ended up with debug statements plus an empty array as a fallback for Release builds. We can throw an exception instead of returning an empty array, but this slows the inspect tool. Is there any reason we should throw an exception for Release builds?